src/vertexcodec.cpp

```diff
@@ -83,9 +83,22 @@
 #endif
 
 #ifdef SIMD_WASM
+// WASM simd updated some macros
+// ie: 'wasm_v32x4_shuffle' is deprecated
+// https://emscripten.org/docs/porting/simd.html
+// https://github.com/llvm/llvm-project/blob/main/clang/lib/Headers/wasm_simd128.h
 #undef __DEPRECATED
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+#pragma clang diagnostic ignored "-W#pragma-messages"
+// byte, enc, encv, data_var
+#pragma clang diagnostic ignored "-Wunused-variable"
```

Comment:
I think this and below would go away if this included WASM in the condition (I think it's an oversight that it doesn't):

```
#if defined(SIMD_FALLBACK) || (!defined(SIMD_SSE) && !defined(SIMD_NEON) && !defined(SIMD_AVX))
static const unsigned char* decodeBytesGroup(const unsigned char* data, unsigned char* buffer, int bitslog2)
```

Reply:
using 
```cpp
#if defined(SIMD_FALLBACK) || (!defined(SIMD_SSE) && !defined(SIMD_NEON) && !defined(SIMD_AVX) && !defined(SIMD_WASM))
static const unsigned char* decodeBytesGroup(const unsigned char* data, unsigned char* buffer, int bitslog2)
```
it allows to remove that part

```cpp
// decodeVertexBlock
#pragma clang diagnostic ignored "-Wunused-function"
```